core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/WrappedSnapshot.scala

```diff
@@ -0,0 +1,23 @@
+package org.broadinstitute.dsde.rawls.snapshot
+
+import bio.terra.datarepo.model.{CloudPlatform, DatasetSummaryModel, SnapshotModel}
+
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+// A basic wrapper class to simplify interaction with the SnapshotModel from TDR
+class WrappedSnapshot(snapshot: SnapshotModel) {
+
+  // While getSource returns a list of SnapshotSourceModels, this is an artifact and we can make an assumption that
+  // a given snapshot will have exactly one source that will always have a dataset.  Violations of this assumption
+  // indicate a bug, in which case, crash and burn.
+  private def snapshotDataset(): DatasetSummaryModel = getOnlyElement(snapshot.getSource.asScala.toList).getDataset
+
+  private def getOnlyElement[A](list: List[A]): A = list match {
+    case head :: Nil => head
```

Comment:
nit: since `head` is a method of `Iterable`, I suggest using a different variable name here to avoid confusion, like:
```suggestion
    case x :: Nil => x
```

Reply:
Fixed this in 941cd9d16b04d2499c031cb68682aee1f9016761, although I try to avoid single-character variable names unless I'm really cramped for space...so I chose `onlyElement`

That said, I don't _actually_, _really_ understand what the variable is referring to here because we're deep in quirky Scala stuff, so if that name doesn't actually make sense please let me know!!

Reply:
Looks like you've got it right. Scala implements `List` as a collection of elements terminated by a `Nil`. If you were working with a `List[String]`, you could have:
| List length | `List` constructor | pattern-match |
|--------|--------|--------|
| 0 | List() | Nil |
| 1  | List("foo") | x :: Nil |
| 2  | List("foo", "bar") | x :: y :: Nil |

you can also pattern-match against `List`s by looking at the head (one element) and tail (N elements). `List("foo", "bar", "baz", "qux")` will pattern match against `x :: y`, where `x` is a `String` of "foo" and `y` is a `List("bar", "baz", "qux")`.

Reply:
🤯